### DISCUSSION TITLE
[SUGGESTION] Unlocking hidden features

### DISCUSSION THREAD

**Discussion by Yehonal** [azerothcore/azerothcore-wotlk#3939] [Ideas]
Hi folks

AzerothCore has a solid blizzlike layer that allows you to recreate the original experience of the expansion itself. 

When you talk about Blizzlike, **AzerothCore is the way to go!**

However, as many of you know, our customizable layer is even better! Modules, our WIP project for DBC as SQL and many other features allow you to create a fully-customizable world to keep the 
gaming experience fresh and clean from the awful core patching. And you can do it with just one click thanks to the docker support and many other stuff!

But it's not only about the system itself, but it's also because we're creating an entire ecosystem around it allowing you to find the right thing to build whatever you want!

So, when you talk about custom, **AzerothCore is DEFINITELY the way to go!**

Today I was thinking about a topic that could be interesting for all of you: 

> **Unlocking the "hidden" features of the client.**

**What?**

There are some hidden features inside the client such as the "black rain" weather, EPIC difficulty for dungeons, hidden maps that can be used to implement custom stuff.

**How?**

Sending particular data via the opcode using certain values from db[c]/code you can unlock testing stuff, hidden maps, unused options etc.

When we need to change something about the code to unlock it, we can implement it as an optional and configurable feature (either in worldserver.conf or as a cmake preprocessor option)

On the basis of such core changes, we all (yes you too) can create modules allowing the creation of entire playable systems adding db data, scripts and even new opcodes handled by game addons if truly necessary.

**Why?**

Because it's fun! Feeling like a not limited gaming artist

**Where?**

I would like to create a list **here** and keep it updated. It can be used to create modules and as said before, when needed, implement the support for such feature as an optional configuration inside the core.

So, let's start to collect them:

- Did you know that there's a black rain and black snow in-game? I discovered it during an investigation on Jan 2008 when I was fixing the mangos weather system: [commit](https://github.com/mangos/mangos-svn/commit/52a48cf844d2b4111e01f6da1591c7e2562b8f35)
My nickname was HW2-dwarf (totally forgot it 😨). You can use it for some role-play stuff or if you want to scare people 😛 

- Did you know that you can learn languages gradually? Setting a language to skill 1, your character can learn/skill it just speaking! There are several not used languages: https://www.azerothcore.org/wiki/Languages A module could be created with some books to start learning them and eventually create levels such as in professions. 

- Did you know you can explore hidden maps? such as the under-working Emerald Dream, the Azshara Crater etc? (some of them have been lost during expansion change, such as the amazing Development Island,  or the [WIP outlands](https://goo.gl/photos/x8VGiVjR7NE7SM7F9) does someone remember it?) maybe a module to teleport in those hidden locations could be interesting. Is there something out of there?

- Did you know you can enable the EPIC difficulty for all dungeons? I've seen a PR [here](https://github.com/azerothcore/azerothcore-wotlk/pull/2958/files#diff-83ab1d973ffb0297b7ff9f81ac1c9ab9R268) But the author mixed it with a mythic+ system. IMHO we should integrate the possibility to enable the difficulty option inside the core but keeping the mythic+ system outside the core as a module

Please, if you know some hidden features or have an idea on how to use the core to unlock them, discuss it below inside the comments and I'll add it to the list!

**Comment by Helias** [comment]
Add this link for Emerald Dream teleports location and the related texture patch.

https://github.com/azerothcore/azerothcore-wotlk/pull/2333

**Comment by BarbzYHOOL** [comment]
How do we use the blackrain? http://www.azerothcore.org/wiki/game_weather this doesn't seem possible here
Maybe there is a command too?

> IMHO we should integrate the possibility to enable the difficulty option inside the core but keeping the mythic+ system outside the core as a module

what? if it's outside, then it's outside, not inside + outside

There are some hidden stuff uncovered by Turtle WOW too and also in another emulator I don't remember the name (the one who first found 2 factor authentification then got copied by everybody else, including us)

**Comment by Yehonal** [comment]
> what? if it's outside, then it's outside, not inside + outside

EPIC difficulty (hidden option) ... Mythic+ (a completely different system that is not related at all with EPIC difficulty). EPIC inside, Mythic+ outside. Clear?

> How do we use the blackrain?

weather change command is not working? you should be able to use the blackrain with the GM command using the correct weather type
